core/frontend/src/components/wifi/ConnectionDialog.vue

```diff
@@ -0,0 +1,178 @@
+          >
+            <v-card-text>
+              Force new password
+            </v-card-text>
+          </v-card>
+          <password-input
+            v-if="show_password_input_box"
+            v-model="password"
+            @submit="connectToWifiNetwork"
+          />
+        </v-row>
+
+        <v-row>
+          <v-col cols="6">
+            <v-btn
+              color="#2c99ce"
+              depressed
+              @click="connectToWifiNetwork"
+            >
+              Connect
+            </v-btn>
+          </v-col>
+
+          <v-col cols="6">
+            <v-btn
+              v-if="network.saved"
+              color="#EF5350"
+              depressed
+              @click="removeSavedWifiNetwork"
+            >
+              Forget
+            </v-btn>
+          </v-col>
+        </v-row>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue'
+import { getModule } from 'vuex-module-decorators'
+import WifiStore from '@/store/wifi'
+import CommonStore from '@/store/common'
+import PasswordInput from '../common/PasswordInput.vue'
+import { Network } from '@/types/wifi'
+import { ErrorIndex } from '@/types/common'
+import axios from 'axios'
+
+const common_store: CommonStore = getModule(CommonStore)
+const wifi_store: WifiStore = getModule(WifiStore)
+
+export default Vue.extend({
+  name: 'ConnectionDialog',
+  model: {
+    prop: 'show',
+    event: 'change',
+  },
+  components: {
+    PasswordInput,
+  },
+  data () {
+    return {
+      password: '',
+      force_password: false,
+      inputed_ssid: '',
+    }
+  },
+  props: {
+    network:  {
+      type: Object as PropType<Network>,
+      required: true,
+    },
+    show: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  computed: {
+    real_ssid (): string {
+      return this.is_hidden ? this.inputed_ssid : this.network.ssid
+    },
+    is_hidden (): boolean {
+      return this.network.ssid === null || this.network.ssid === ''
+    },
+    show_password_input_box (): boolean {
+      if (this.force_password) {
+        return true
+      }
+      else if (this.network.saved || !this.network.locked) {
+        return false
+      }
+      return true
+    },
+  },
+  methods: {
+    async connectToWifiNetwork (): Promise<void> {
+      const password = this.password
+      this.password = ''
+      this.emitChange(false)
+
+      const connect_response = await axios({
+        method: 'post',
+        url: `${wifi_store.API_URL}/connect`,
+        timeout: 500,
+        data: {ssid: this.real_ssid, password},
+        params: { hidden: this.is_hidden },
+      })
+
+      if (connect_response.status != 200) {
+        common_store.openError({index: ErrorIndex.WifiConnectFail, message: 'Could not connect to wifi network.'})
+        return
+      }
+      common_store.closeError(ErrorIndex.WifiConnectFail)
+
+      wifi_store.setNetworkStatus(null)
```

Comment:
What is a null state ? This should be more clear.

Reply:
It's a `null` only when there's no status information available, as in the case of API request errors.